README.md

```diff
@@ -21,156 +41,103 @@ true
 >
 > ```elixir
 > iex> Date.compare(~D[2017-03-31], ~D[2017-04-01])
 > :lt
 > ```
 
-In other words, although `~D[2017-03-31] > ~D[2017-04-01]` is perfectly valid code, it does _not_ tell you if `~D[2017-03-31]` is a later date than `~D[2017-04-01]` like you might expect.
-Instead, you need to use `Date.compare/2`.
+In other words, although `~D[2017-03-31] > ~D[2017-04-01]` is valid code, it does _not_ tell you if `~D[2017-03-31]` is a later date than `~D[2017-04-01]` as you might expect.
+Instead, you'd use `Date.compare/2`.
 
-### Chained vs. nested comparisons
+### Chained vs. Nested Comparisons
 
-Additionally, even if `~D[2017-03-31] > ~D[2017-04-01]` did do semantic comparison, you still couldn't write the interval check like you do in Python.
-This is because in Python, an expression like `1 < 2 < 3` is syntactic sugar for `(1 < 2) and (2 < 3)`, aka a series of "chained" expressions.
+Additionally, even if `~D[2017-03-31] > ~D[2017-04-01]` did semantic comparison, you still couldn't write the interval check like you do in Python. In Python, an expression like `1 < 2 < 3` is syntactic sugar for `(1 < 2) and (2 < 3)` (a series of "chained" expressions).
 
-Elixir does not provide an equivalent syntactic sugar.
-Instead, `1 < 2 < 3` is evaluated as `(1 < 2) < 3`, aka a series of "nested" expressions.
-Since `(1 < 2) < 3` simplifies to `true < 3`, that's probably not what you want!
+Elixir doesn't provide an equivalent syntactic sugar. Instead, `1 < 2 < 3` is evaluated as `(1 < 2) < 3` (a series of "nested" expressions). `(1 < 2) < 3` evaluates to `true < 3` which is _probably_ not what you want!
 
-Elixir will even warn you when you attempt an expression like that:
+### A Solution!
 
-> warning: Elixir does not support nested comparisons. Something like
->
->      x < y < z
->
-> is equivalent to
->
->      (x < y) < z
->
-> which ultimately compares z with the boolean result of (x < y). Instead, consider joining together each comparison segment with an "and", for example,
->
->      x < y and y < z
-
-### CompareChain
-
-`CompareChain` attempts to address both of these issues with the macro `CompareChain.compare?/2`.
-Its job is to take code similar to how you'd like to write it and rewriting it to be semantically correct.
-
-For our motivating example, we'd write this:
+CompareChain addresses these complexities with the macro `CompareChain.compare?/2`:
 
 ```elixir
 import CompareChain
-compare?(start_date <  date <  end_date, Date) # excluding bounds
-compare?(start_date <= date <= end_date, Date) # including bounds
+
+# excluding bounds
+compare?(start_date < date < end_date, Date)
+
+# including bounds
+compare?(start_date <= date <= end_date, Date)
 ```
 
-And at compile time, `CompareChain.compare?/2` rewrites those to be:
+`CompareChain.compare?/2` compiles these expressions as:
```

Comment:
```suggestion
`CompareChain.compare?/2` rewrites these expressions as:
```

I think it's incorrect to say `CompareChain.compare?/2` compiles anything. From the [docs](https://hexdocs.pm/elixir/Macro.html):

> Macros are compile-time constructs that receive Elixir's AST as input and return Elixir's AST as output.

So it's code that is executed at compile time, but it does no compilation itself. This is admittedly a _pretty_ nit-picky point but I don't want to misrepresent what the library is doing.